--- Sym-Base.mf.orig	2006-05-23 00:48:58.000000000 +0200
+++ Sym-Base.mf	2006-05-23 00:49:30.000000000 +0200
@@ -460,3 +460,120 @@
   penlabels(0$,1$,2$,3$,4$,5$,6$);
 enddef;
 
+def draw_arrowhead(expr pos, angle, spread) =
+  begingroup
+    save z; pair z[];
+    save width;
+    width = rule_thickness;
+    z00 = pos;
+    z10 = pos - 3/4spread * dir angle + spread/2 * dir (angle + 90);
+    z20 = pos - 1/4spread * dir angle;
+    z30 = pos - 3/4spread * dir angle + spread/2 * dir (angle - 90);
+    %%%%% z00
+    z02 = z00 + width/2 * dir (angle -  80);
+    z05 = z00 + width/2 * dir  angle;
+    z08 = z00 + width/2 * dir (angle +  80);
+    %%%%% z10
+    z12 = z10 + width/2 * dir (angle +  40);
+    z15 = z10 + width/2 * dir (angle + 130);
+    z18 = z10 + width/2 * dir (angle + 220);
+    %%%%% z20
+    z25 = z20 - width/(2sind(50)) * dir  angle; % approximation!
+    %%%%% z30
+    z32 = z30 + width/2 * dir (angle - 220);
+    z35 = z30 + width/2 * dir (angle - 130);
+    z38 = z30 + width/2 * dir (angle -  40);
+    %%% final path
+    fill z05 {dir (angle + 90)}
+    .. z08 {dir (angle + 170)}
+    .. z12 {dir (angle + 130)}
+    .. z15 {dir (angle + 220)}
+    .. z18 {dir (angle + 310)}
+    .. {dir (angle + 310)} z25 {dir (angle - 130)} % approximation!
+    .. z32{dir (angle - 130)}
+    .. z35{dir (angle -  40)}
+    .. z38{dir (angle - 310)}
+    .. z02{dir (angle +  10)}
+    .. cycle;
+  endgroup
+enddef;
+
+def draw_arrowhead_left(expr pos, angle, spread) =
+  begingroup
+    save z; pair z[];
+    save width;
+    width = rule_thickness;
+    z00 = pos;
+    z10 = pos - 3/4spread * dir angle + spread/2 * dir (angle + 90);
+    z20 = pos - 1/4spread * dir angle;
+    z30 = pos - 3/4spread * dir angle + spread/2 * dir (angle - 90);
+    %%%%% z00
+    z02 = z00 + width/2 * dir (angle -  80);
+    z05 = z00 + width/2 * dir  angle;
+    z08 = z00 + width/2 * dir (angle +  80);
+    %%%%% z10
+    z12 = z10 + width/2 * dir (angle +  40);
+    z15 = z10 + width/2 * dir (angle + 130);
+    z18 = z10 + width/2 * dir (angle + 220);
+    %%%%% z20
+    z25 = z20 - width/(2sind(50)) * dir  angle; % approximation!
+    %%%%% z30
+    z32 = z30 + width/2 * dir (angle - 220);
+    z35 = z30 + width/2 * dir (angle - 130);
+    z38 = z30 + width/2 * dir (angle -  40);
+    %%% final path
+    fill z05 {dir (angle + 90)}
+    .. z08 {dir (angle + 170)}
+    .. z12 {dir (angle + 130)}
+    .. z15 {dir (angle + 220)}
+    .. z18 {dir (angle + 310)}
+    .. {dir (angle + 310)} z25 {dir (angle - 130)} % approximation!
+%    .. z32{dir (angle - 130)}
+%    .. z35{dir (angle -  40)}
+%    .. z38{dir (angle - 310)}
+%    .. z02{dir (angle +  10)}
+    -- cycle;
+  endgroup
+enddef;
+
+
+def draw_arrowhead_right(expr pos, angle, spread) =
+  begingroup
+    save z; pair z[];
+    save width;
+    width = rule_thickness;
+    z00 = pos;
+    z10 = pos - 3/4spread * dir angle + spread/2 * dir (angle + 90);
+    z20 = pos - 1/4spread * dir angle;
+    z30 = pos - 3/4spread * dir angle + spread/2 * dir (angle - 90);
+    %%%%% z00
+    z02 = z00 + width/2 * dir (angle -  80);
+    z05 = z00 + width/2 * dir  angle;
+    z08 = z00 + width/2 * dir (angle +  80);
+    %%%%% z10
+    z12 = z10 + width/2 * dir (angle +  40);
+    z15 = z10 + width/2 * dir (angle + 130);
+    z18 = z10 + width/2 * dir (angle + 220);
+    %%%%% z20
+    z25 = z20 - width/(2sind(50)) * dir  angle; % approximation!
+    %%%%% z30
+    z32 = z30 + width/2 * dir (angle - 220);
+    z35 = z30 + width/2 * dir (angle - 130);
+    z38 = z30 + width/2 * dir (angle -  40);
+    %%% final path
+    fill %z05 {dir (angle + 90)}
+%    .. z08 {dir (angle + 170)}
+%    .. z12 {dir (angle + 130)}
+%    .. z15 {dir (angle + 220)}
+%    .. z18 {dir (angle + 310)}
+%    .. {dir (angle + 310)} 
+    z25 {dir (angle - 130)} % approximation!
+    .. z32{dir (angle - 130)}
+    .. z35{dir (angle -  40)}
+    .. z38{dir (angle - 310)}
+    .. z02{dir (angle +  10)}
+    .. z05 {dir (angle + 90)}
+    -- cycle;
+  endgroup
+enddef;
+
